LICENSE.txt

```diff
@@ -1,84 +1,188 @@
+  the libscheme license. Code from libscheme can be found in
+  "racket/src/bc/src".
+
+* GNU Lightning
+  Copyright © 1994, 1995, 1996, 1999, 2000, 2001, 2002, 2011 Free
+  Software Foundation, Inc.
+  GNU Lightning is distributed under the GNU Lesser General Public
+  License, version 3, or (at your option) any later version.
+  A fork of GNU Lightning can be found in "racket/src/bc/src/lightning".
+
+* GNU MP Library
+  Copyright © 1991, 1992, 1993, 1994, 1996, 1999, 2000, 2007 Free
+  Software Foundation, Inc.
+  GMP is distributed under the GNU Lesser General Public License,
+  version 3, or (at your option) any later version.
+  Code from GMP can be found in "racket/src/bc/src/gmp".
+
+* libunwind
+  Copyright © 2003–2005 Hewlett-Packard Development Company, L.P.
+  libunwind is distributed under the MIT license.
+  Code from libunwind can be found in "racket/src/bc/src/unwind".
+
+* libffi
+  Copyright © 1996–2019 Anthony Green, Red Hat, Inc and others.
+  libffi is distributed under the MIT license.
+  A copy of libffi can be found in "racket/src/bc/foreign/libffi".
+
+* random.inc
+  Based on "random.c" from FreeBSD 2.2.
+  Copyright © 1983, 1993 The Regents of the University of California.
+  This code is distributed under a three-clause BSD license.
+  Code from FreeBSD 2.2 can be found in "racket/src/bc/src/random.inc".
+
+In addition to the Racket runtime system, the default mode of building
+Racket will install some packages, which are distributed under their own
+licenses.
+
+The Racket code for all packages in the main Racket distribution is
+primarily under the MIT license and the Apache License, version 2.0, at
+your option, except that some packages contain code under other
+permissive licenses, and the "string-constants-lib-lgpl" package (which
+provides Spanish, Korean, and Dutch GUI translations) is distributed
+under the GNU Lesser General Public License, version 2.0, or (at your
+option) any later version.
+
+Some components of Racket are implemented using additional shared
+libraries. For some platforms and installation modes (especially for
+Windows and Mac OS), these libraries are distributed in
+platform-specific Racket packages; in other cases, they may be included
+with the operating system or installable via the system’s package
+manager. In all cases, these libraries are dynamically loaded and can be
+replaced with modified variants by users. In particular:
+
+* Even the “minimal Racket” distribution may include shared libraries
+  for OpenSSL, SQLite, and/or Editline, depending on the platform.
+
+  The Windows distribution also includes an iconv
+  implementation—currently GNU libiconv, which is distributed under the
+  GNU Lesser General Public License, either version 3 of the license, or
+  (at your option) any later version. Other platforms use the system’s
+  iconv implementation. See the byte converter documentation for further
+  details.
+
+* The Racket Drawing Toolkit is implemented using Cairo, Pango,
```

Comment:
I was conflicted about this because this set of libraries has an especially intricate set of licenses. I guess it could be summarized that Pango, FriBidi, Glib, and some other things are `LGPL-2.1-or-later`, Cairo is `(LGPL-2.1-only OR MPL-1.1)`, and then there are a bunch of permissive licenses. Aside from adding complexity here, the other downside seemed to be that the more detail we add here, the more chance there is for it to get out of sync with the actual contents of the platform-specific packages.

Maybe some very broad statement could be useful? e.g. "Some of these libraries are distributed under variants of the GNU Lesser General Public License." I think the main message is that, if Cairo is a problem for you, so is `racket/draw` (short of implementing an alternate backend).

Reply:
Right, I was thinking of that kind of broad statement.

Reply:
Done.